### PR TITLE
Allow to include the current version in history table

### PR DESCRIPTION
**Pull request by mcesar**

### CODE REVIEW

**Comment by dekiesel** [comment]
I'd be pretty interested in this too since this makes it much easier to see current and historic data at a glance.

**Comment by simoneb** [comment]
@mcesar can you please explain what this PR does and the motivation why you would need this?

**Comment by mcesar** [comment]
@simoneb this PR gives the option to store the current version in history table together with the old versions. When applied, it will provide a easier way to query data given a timestamp. For example, given a table `t(id integer, name text, sys_period tstzrange)`, we want to query the version at a timestamp x:

a. current behavior:

```sql
SELECT * FROM t where x <@ sys_period
UNION
SELECT * FROM t_history where x <@ sys_period;
```

a. with `include_current_version_in_history=true`:

```sql
SELECT * FROM t_history where x <@ sys_period;
```

**Comment by simoneb** [comment]
Considering that this has a simple solution in user space and that it would deviate from the original implementation that we're trying to adhere to, I don't see a compelling reason to include this feature, but happy to be convinced otherwise

**Comment by simoneb** [comment]
It's not an issue per-se, but this has a very simple solution in user space

**Comment by dekiesel** [comment]
As the number of tables grows so does the number of workarounds.

I am versioning 21 tables and I have to create 21 views when stepping through historic data. This PR would simplify my life by a lot.

**Comment by simoneb** [comment]
If there is demand for this feature, I'm not strongly against including it. We just need to make sure that, if possible, this doesn't result in a breaking change for existing users, and that the default will remain that of not including the current version in the history table.

**Comment by rohaldb** [comment]
This I would find this particularly useful. Running the union for me is prohibitively expensive at inference time, meaning I then have to introduce a delay to do it periodically using cron.

**Comment by leventea1** [comment]
@mcesar can you update `versioning_function_nochecks.sql `and document it in Readme as well.

**Review by leventea1 — COMMENTED** [review]
lgtm

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,5 @@
 *.tmp
 .DS_Store
-test/result
\ No newline at end of file
+test/result
```
Comment (simoneb) [inline .gitignore:3]: this is a test to check that I can commit changes to this branch

Comment (simoneb) [inline .gitignore:3]: test

```suggestion
test/results
```

**Comment by kieranswhite** [comment]
@mcesar @dekiesel @rohaldb I am going to close this PR as we have taken changes proposed by @mcesar and merged them in to the extension along with a few small additions. It's been released under version [v0.6.0](https://github.com/nearform/temporal_tables/releases/tag/v0.6.0). Thanks all!

**Pull request closed without merging**